libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/speedlimit/CarSpeedLimitRenderer.kt

```diff
@@ -46,39 +52,51 @@ class CarSpeedLimitRenderer(
         }
     }
 
+    private lateinit var scope: CoroutineScope
+
     private fun updateSpeed(locationMatcherResult: LocationMatcherResult) {
         val speedKmph =
             locationMatcherResult.enhancedLocation.speed / METERS_IN_KILOMETER * SECONDS_IN_HOUR
+        val speedLimitOptions = mainCarContext.speedLimitOptions.value
+        val signFormat = speedLimitOptions.forcedSignFormat
+            ?: locationMatcherResult.speedLimit?.speedLimitSign
+        val threshold = speedLimitOptions.warningThreshold
         when (mainCarContext.mapboxNavigation.navigationOptions.distanceFormatterOptions.unitType) {
             UnitType.IMPERIAL -> {
                 val speedLimit =
                     locationMatcherResult.speedLimit?.speedKmph?.let { speedLimitKmph ->
                         5 * (speedLimitKmph / KILOMETERS_IN_MILE / 5).roundToInt()
                     }
                 val speed = speedKmph / KILOMETERS_IN_MILE
-                speedLimitWidget?.update(speedLimit, speed.roundToInt())
+                speedLimitWidget?.update(speedLimit, speed.roundToInt(), signFormat, threshold)
             }
             UnitType.METRIC -> {
-                speedLimitWidget?.update(
-                    locationMatcherResult.speedLimit?.speedKmph,
-                    speedKmph.roundToInt()
-                )
+                val speedLimit = locationMatcherResult.speedLimit?.speedKmph
+                speedLimitWidget?.update(speedLimit, speedKmph.roundToInt(), signFormat, threshold)
             }
         }
     }
 
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
         logAndroidAuto("CarSpeedLimitRenderer carMapSurface loaded")
-        val speedLimitWidget = SpeedLimitWidget().also { speedLimitWidget = it }
+        val forcedSignFormat = mainCarContext.speedLimitOptions.value.forcedSignFormat
+        val speedLimitWidget = SpeedLimitWidget(
```

Comment:
Can we lazily create an instance of `SpeedLimitWidget` once the speed is available? That way there will not be a flicker of speed limit signs

Reply:
I wasn't able to reproduce such flicker. Locally I initialized the widget with MUTCD sign type, but changed to Vienna in location observer. And looks like the widget doesn't get a chance to be drawn using MUTCD style. On the other hand lazy creation complicates the code, because I have to save `MapboxCarMapSurface` and visible area into properties, so that they can be used when the widget is created later.

Reply:
i'm thinking there will be a flicker under certain scenarios. it may be hard to reproduce when caches are loaded or the network is fast.

but real life for driving cars will typically has slow network, and fresh installs happen often (5% case?).

in my opinion, should still move it to lazy because then it will be impossible. but am leaving up to you

Reply:
Why will it be impossible later? I don't think making this change will break API.

Reply:
This will be easier to implement once mapbox/mapbox-maps-android#1531 is fixed, because we will be able to simply initialize the widget with the empty bitmap.